### PR TITLE
add code generator for Opentype tables parser

### PR DESCRIPTION
**Pull request by benoitkugler**
This PR is the second requirement for the (incoming) Opentype parser.

It consists of a code generator which reads Go struct declarations and generates parsing functions (converters from `[]byte` to a Go struct).
It is smart enough to handle usual constructs in Opentype files, such as arrays and offsets.
When the table format is not supported by this generator, one can fallback to a hand written function.

You can see the generated code for instance [here](https://github.com/benoitkugler/go-opentype/blob/main/tables/glyphs_bitmap_gen.go) .

The main benefit of this approach is to potentially make the reverse operation (in memory Go structs to binary format) much more easy (although I do not plan to work on it soon).

One could argue that this tool belongs to an external module, but :

1. The generator is totally tied to the Opentype format and conventions, and barely usable in other contexts.
2. It seems desirable to group all the dev dependencies in one place (with no cost for end users, thanks to module pruning).
3. Other code generators will be added for Unicode data and Harfbuzz Unicode properties, so that it wont be an exception.

### CODE REVIEW

File: generators/binarygen/analysis/analysis.go [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+// shared by the parser and writer code generator
+type Analyser struct {
+	// Source is the path of the origin go source file.
+	Source string
+
+	sourceAbsPath string
+
+	pkg *packages.Package
+
+	// Tables contains the resolved struct definitions, coming from [Sources]
```
Comment (whereswaldon) [inline generators/binarygen/analysis/analysis.go]: What is `[Sources]` in this context? The closest matching field is `Source`. Is this perhaps leftover from a refactor?

Comment (benoitkugler) [reply generators/binarygen/analysis/analysis.go]: It was a leftover indeed, thank you. I've fixed the comment.

**Review by andydotxyz — APPROVED** [review]
Pretty epic :)

File: generators/binarygen/analysis/tags.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+type commments struct {
+	// externalArguments may be provided it the type parsing/writting function
+	// requires data not provided in the input slice
+	externalArguments []Argument
+}
+
+// parse the type documentation looking for special comments
+// of the following form :
```
Comment (whereswaldon) [inline generators/binarygen/analysis/tags.go:155]: Looks like this comment is missing an example comment that demonstrates the formatting.

**Review by whereswaldon — APPROVED** [review]
What a colossal and impressive effort! Thank you. :D

File: generators/binarygen/analysis/tags.go (hunk 2 above)
Comment (benoitkugler) [reply generators/binarygen/analysis/tags.go:155]: Indeed, thank you. Fixed.

**Pull request merged**